### PR TITLE
[CPPRT][CRT][MSVCRT] Move __CxxFrameHandler3 to crt and export it on NT6

### PR DESCRIPTION
**Pull request by tkreuzer**
Previously it was in cpprt, which is a support library for C++, containing functions that are not exported by msvcrt. But since Vista __CxxFrameHandler3 is exported by msvcrt. Therefore move it to crt, and to satisfy pre-Vista configurations, also add it to msvcrtex.

### CODE REVIEW

File: sdk/lib/crt/except/amd64/ehandler.c [hunk 1]
```diff
@@ -125,8 +123,8 @@ __C_specific_handler(
                             DispatcherContext->HistoryTable);
 
                 /* Should not get here */
-                __debugbreak();
-            }
+    __debugbreak();
+}
```
Comment (SergeGautherie) [inline sdk/lib/crt/except/amd64/ehandler.c:127]: Unwanted change?

File: sdk/lib/crt/except/amd64/ehandler.c [hunk 2]
```diff
@@ -8,8 +8,6 @@
 #include <precomp.h>
 #include <winnt.h>
 
-
-_CRTIMP
```
Comment (SergeGautherie) [inline sdk/lib/crt/except/amd64/ehandler.c:12]: Unwanted change?

File: sdk/lib/crt/msvcrtex.cmake [hunk 3]
```diff
@@ -63,6 +69,11 @@ elseif(ARCH STREQUAL "arm")
         math/arm/__rt_srsh.s
         math/arm/__rt_udiv64.s
     )
+    if(MSVC AND DLL_EXPORT_VERSION LESS 0x600)
+        list(APPEND MSVCRTEX_ASM_SOURCE
+            except/amd64/__CxxFrameHandler3.s
+        )
+    endif()
```
Comment (SergeGautherie) [inline sdk/lib/crt/msvcrtex.cmake:76]: Move it to `amd64` block above.

File: sdk/lib/crt/except/except.cmake [hunk 4]
```diff
@@ -4,9 +4,13 @@ if(ARCH STREQUAL "i386")
         except/i386/chkstk_asm.s
     )
     list(APPEND CRT_EXCEPT_ASM_SOURCE
+        except/i386/__CxxFrameHandler3.s
         except/i386/chkesp.s
         except/i386/prolog.s
     )
+    list(APPEND CRT_EXCEPT_SOURCE
+        except/i386/CxxHandleV8Frame.c
+    )
```
Comment (SergeGautherie) [inline sdk/lib/crt/except/except.cmake:13]:
```suggestion
        except/i386/CxxHandleV8Frame.c)
```

File: sdk/lib/crt/msvcrtex.cmake [hunk 5]
```diff
@@ -36,9 +36,20 @@ if(ARCH STREQUAL "i386")
...
+        list(APPEND MSVCRTEX_SOURCE
+            except/i386/CxxHandleV8Frame.c)
+    endif()
 elseif(ARCH STREQUAL "amd64")
     list(APPEND MSVCRTEX_ASM_SOURCE
         except/amd64/chkstk_ms.s)
+    if(MSVC AND DLL_EXPORT_VERSION LESS 0x600)
+        list(APPEND MSVCRTEX_ASM_SOURCE
+            except/amd64/__CxxFrameHandler3.s
+        )
```
Comment (SergeGautherie) [inline sdk/lib/crt/msvcrtex.cmake:51]:
```suggestion
            except/amd64/__CxxFrameHandler3.s)
```

File: sdk/lib/crt/except/except.cmake (hunk 4 above)
Comment (tkreuzer) [reply sdk/lib/crt/except/except.cmake:13]: no

Comment (SergeGautherie) [reply sdk/lib/crt/except/except.cmake:13]: Note:
I don't really mind either way, I am simply wondering why some parts have the parenthesis on the same line and some on the next line.

File: sdk/lib/crt/except/i386/__CxxFrameHandler3.s [hunk 6]
```diff
@@ -0,0 +1,26 @@
+
+
```
Comment (HBelusca) [inline sdk/lib/crt/except/i386/__CxxFrameHandler3.s:2]:
```suggestion
```

File: sdk/lib/crt/except/i386/__CxxFrameHandler3.s [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+#include <asm.inc>
+
+.code
+.align 4
```
Comment (HBelusca) [inline sdk/lib/crt/except/i386/__CxxFrameHandler3.s:6]: What does this ".align 4" affect, in this code?

Comment (tkreuzer) [reply sdk/lib/crt/except/i386/__CxxFrameHandler3.s:6]: The alignment.

**Pull request merged**